src/native/eventpipe/ep-buffer-manager.c

```diff
@@ -897,12 +900,23 @@ ep_buffer_manager_write_event (
 	// Before we pick a buffer, make sure the event is enabled.
 	ep_return_false_if_nok (ep_event_is_enabled (ep_event));
 
+	// Check that the payload size is less than 64 KB (max size for ETW events)
+	if (ep_event_payload_get_size (payload) > 64 * 1024)
+	{
+		ep_rt_atomic_inc_int64_t (&buffer_manager->num_oversized_events_dropped);
```

Comment:
Other, similar counters on `EventPipeBufferManager` are only compiled in for checked builds. Do we think there will be enough diagnostic benefit to include this operation in release builds as well?

Reply:
At this point I think it'd be better if we come up with a way to make EventPipe more diagnosable by adding more stress logs, adding more of these backed metrics, etc and have that as a separate PR.